olive/azureml/azureml_client.py

```diff
@@ -89,38 +82,59 @@ def create_client(self):
         logging.getLogger("azure.identity").setLevel(logging.ERROR)
 
         if self.aml_config_path is None:
+            if self.subscription_id is None:
+                raise ValueError("subscription_id must be provided if aml_config_path is not provided")
+            if self.resource_group is None:
+                raise ValueError("resource_group must be provided if aml_config_path is not provided")
+            if self.workspace_name is None:
+                raise ValueError("workspace_name must be provided if aml_config_path is not provided")
             return MLClient(
-                credential=self._get_credentials(),
+                credential=_get_credentials(self.default_auth_params),
                 subscription_id=self.subscription_id,
                 resource_group_name=self.resource_group,
                 workspace_name=self.workspace_name,
                 read_timeout=self.read_timeout,
             )
         else:
             return MLClient.from_config(
-                credential=self._get_credentials(), path=self.aml_config_path, read_timeout=self.read_timeout
+                credential=_get_credentials(self.default_auth_params),
+                path=self.aml_config_path,
+                read_timeout=self.read_timeout,
             )
 
-    def _get_credentials(self):
-        """Get credentials for MLClient.
-
-        Order of credential providers:
-        1. Azure CLI
-        2. DefaultAzureCredential
-        3. InteractiveBrowserCredential
-        """
-        from azure.identity import DefaultAzureCredential, InteractiveBrowserCredential
-
-        logger.debug("Getting credentials for MLClient")
-        try:
-            default_auth_params = self.default_auth_params or {}
-            credential = DefaultAzureCredential(**default_auth_params)
-            # Check if given credential can get token successfully.
-            credential.get_token("https://management.azure.com/.default")
-            logger.debug("Using DefaultAzureCredential")
-        except Exception:
-            # Fall back to InteractiveBrowserCredential in case DefaultAzureCredential not work
-            credential = InteractiveBrowserCredential()
-            logger.debug("Using InteractiveBrowserCredential")
-
-        return credential
+    @staticmethod
+    def create_registry_client(registry_name: str, **kwargs):
```

Comment:
>kwargs [](http://example.com/codeflow?start=53&length=6)

it seems the kwargs only use default_auth_params? if yes, we could make the parameter explicitly